File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        Stack<Character> parenthesis = new Stack<>();
        
        for(int i = 0; i < s.length(); i++) {
            char curr = s.charAt(i);
            if (curr == '(') {
                parenthesis.add(curr);
            }
            else {
                if (parenthesis.isEmpty()) {
                    return false;
                }
                parenthesis.pop();
            } 
        }

        if (parenthesis.isEmpty()) {
            return true;
        }
        
        return false;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <stack>

using namespace std;

bool solution(string s)
{
    
    stack<char> stack;
    for (auto & ch : s) {
        if (ch == '(') {
            stack.push(ch);
        }
        else if (!stack.empty()) {
            stack.pop();
        }
        else {
            return false;
        }
    }
    
    
    
    return stack.empty() ? true : false;
}